Project/core/src/main/java/Model/Board.java

```diff
@@ -0,0 +1,53 @@
+package Model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The Board class represents the game board, which consists of a grid of cells.
+ * Each cell can hold various features that affect gameplay.
+ */
+public class Board {
+
+    private Cell[][] cells; // 2D array representing the grid of cells on the board
+
+    /**
+     * Constructor to initialize the game board with a specified number of rows and
+     * columns.
+     * It creates a grid of cells and initializes them.
+     * 
+     * @param rows    The number of rows in the game board.
+     * @param columns The number of columns in the game board.
+     */
+    public Board(int rows, int columns) {
+        cells = new Cell[rows][columns]; // Initialize the 2D array of cells
+        createCells(); // Create and initialize the cells with features
+    }
+
+    /**
+     * Creates and initializes each cell on the board.
+     * Each cell is initialized with an empty list of features.
+     * TODO: In the future, the cells should be filled with features somehow:
+     * - Either in the createCells method
+     * - Or it's done with a new Cell.addFeatures() method
+     */
+    private void createCells() {
+        List<Feature> features = new ArrayList<>(); // List to hold features for each cell
```

Comment:
wont all cells share the features list if defined in this way? so that if I add one feature to a cell, all cells get the feature...
I might be wrong

Reply:
I think so. Cause this list gets associated with every cell. We should instead initialize a new array for each cell in the forloop. Alternatively just using a tuple (or java equivalent) since each cell only needs two features. So line 38 would instead be "cells[row][column] = new Cell((feature1, feature2))" with feature1 and 2 being two variables each containing a Feature. This however would not need be unique since every feature should be static. It should basically just be an identifier.

Reply:
Good catch Filip! That is very true, we are initializing it at the wrong spot, all cells currently "point" to the same list which is wrong!

I like your idea Fox, tuples does not exist in Java, but maybe we could initialize an array of static size 2?